pkg/agent/controller/egress/egress_controller.go

```diff
@@ -959,3 +959,12 @@ func (c *EgressController) GetEgress(ns, podName string) (string, error) {
 func isEgressSchedulable(egress *crdv1a2.Egress) bool {
 	return egress.Spec.EgressIP != "" && egress.Spec.ExternalIPPool != ""
 }
+
+// GetEgressIPByName returns the Egress IP associated with the Egress.
+func (c *EgressController) GetEgressIPByName(egress string) (string, error) {
+	state, exists := c.getEgressState(egress)
+	if !exists {
+		return "", fmt.Errorf("no Egress State associated with name %s", egress)
+	}
+	return state.egressIP, nil
+}
```

Comment:
Sure I can merge them certainly. I didn't do this in the beginning because had concerns about affecting Traceflow. 
May I ask how it could remove a lock acquring by merging? Since we acquired the `egressBindingsMutex` to obtain the egress name and acquire the `egressStatesMutex` to retrieve the egress IP. Is one of the lock could be skipped? Thanks!

Reply:
Ah, I thought they use the same lock. It can't save a lock acquring then. But I still think it's simpler to merge the two methods given they return two attributes of the same object. And Traceflow queries are not very frequent so not a big deal.

Reply:
Got it, no problem I did the merge.